### DISCUSSION TITLE
Listen to next and previous button clicks of mat-datepicker with out custom header

### DISCUSSION THREAD

**Discussion by skota524** [angular/components#25172] [Ideas]
**feature request or proposal:**
mat-datepicker is really doing great job, but i would like to have a feature where we can listen to next and previous button clicks.

**What is the expected behavior?**

Emit those events from calendar header and from date picker so that we can listen from our component.

calendar-header > emit > date picker > emit > user component 

**What is the current behavior?**

Need to override the calendar-header just for one functionality

**What is the use-case or motivation for changing an existing behavior?**

On date change, we need to do some validations and apply style to the dates.

**Comment by Lertis** [comment]
Any updates for this feature?

**Comment by Lertis** [comment]
Currently, we found two ways to handle in manually. @skota524 maybe it may help you
1 -  Subscribe to the next/previous header buttons click
![querySelector](https://user-images.githubusercontent.com/24374130/99049015-fe832480-259e-11eb-9a6f-47a05f5188f8.jpg)
2 - Update needed styles using Subject +distinctUntilChanged()
![image](https://user-images.githubusercontent.com/24374130/99049109-1fe41080-259f-11eb-8b9b-8a658455e28b.png)

[a comment by an automated account is not shown]

**Comment by zarend** [comment]
Hello, this feature is not in our roadmap, but it is still open for discussion.
